File: Source/Tool/djb2.c
#include "djb2.h"

uint32_t djb2(const char* str)
{
	uint32_t hash = 5381;
	int32_t c = 0;

	while (c = *str++)
		hash = ((hash << 5) + hash) + c; /* hash * 33 + c */

	return hash;
}

File: Source/djb2_.py
def djb2(text):
    hash_value = 5381
    for char in text:
        hash_value = ((hash_value << 5) + hash_value) + ord(char)
    return hash_value & 0xFFFFFFFF

while True:
    user_text = input("Input text: ")
    user_size = int(input("Input table size: "))
    hash = djb2(user_text) 
    print("Hash: " + str(hash))
    print(str(hash) + " % " + str(user_size) + " = " + str(hash % user_size))


print("Ended program.")